### ISSUE TITLE
Function declarations dont follow style guide any longer

### ISSUE THREAD

**Issue by gitpusha** [prettier-solidity/prettier-plugin-solidity#813]
I noticed that as of newer versions of the plugin (I recently switched to `1.1.2` from a previous `1.0.0-dev.23` version), `function declarations` formatting is not in accordance with the [Solidity Style Guide](https://docs.soliditylang.org/en/v0.8.19/style-guide.html#function-declaration) any longer.

The style guide wants this (and the plugin used to format function declarations accordingly):

![telegram-cloud-photo-size-2-5474304915812761941-y](https://user-images.githubusercontent.com/36712489/222104687-b5d554be-aa5a-4717-bcfd-07b0f1769ba8.jpg)

I noticed that the new plugin version I am using ( prettier-plugin-solidity: 1.1.2  ) reformatted all my solidity code and led to a lot of new function declaration formatting as you can see here with `setWithdrawalLockPeriod` 

![telegram-cloud-photo-size-2-5474304915812761944-y](https://user-images.githubusercontent.com/36712489/222104828-868642df-956c-4070-80b5-d031a30503b7.jpg)

Or here with `transfer`:

![telegram-cloud-photo-size-2-5474304915812761945-y](https://user-images.githubusercontent.com/36712489/222104901-3ff82d28-218e-4b2b-b186-d31aefadd726.jpg)

Basicallly it doesnt follow this style guide recommendation any longer: 

> For long function declarations, it is recommended to drop each argument onto its own line at the same indentation level as the function body.

Given that this behavior of `prettier-plugin-solidity` was introduced in recent releases, it might be a regression. 

Would be great to have the old Solidity style-guide compliant `function declaration` formatting back!

**Comment by fvictorio** [comment]
Hey @gitpusha, we don't really have a goal of following the style guide. There are probably other things we are already doing in a different way.

How to format function signatures was by far the hardest thing to decide for v1.0.0. We decided on this approach in part because of our personal preferences, in part by getting feedback from some developers. This is of course super subjective and we know some people won't like it, but now that we've released the stable 1.0.0 it's unlikely that we'll change it. Maybe in the next major :sweat_smile:
